### PR TITLE
Make symlinks so that RBE will work for Xcode toolchains

### PR DESCRIPTION
**Pull request by dnfield**
See https://github.com/flutter/flutter/issues/72032

This creates symlinks for the macOS SDK, iPhone SDK, and Xcode toolchain sysroot in the gen dir. That way, RBE won't fail when it tries to access a file based on an absolute path to /Applications/whatever on mac based host/iOS builds.

### CODE REVIEW

**Comment by zanderso** [comment]
Do the symlinks need to go in a .gitignore?

**Comment by dnfield** [comment]
They're created in a generated out directory - e.g. `out/host_debug_unopt/gen/MacOSX11.0.sdk`.

File: build/mac/find_sdk.py [hunk 1]
```diff
@@ -10,15 +10,25 @@
...
 import os
 import re
 import subprocess
 import sys
 
-
 from optparse import OptionParser
 
 
+def mkdir_p(path):
```
Comment (zanderso) [inline build/mac/find_sdk.py:22]: Is there any way to avoid copying this to several places?

Comment (dnfield) [reply build/mac/find_sdk.py:22]: Yeah. I also realize that I need to fix this so it replaces the symlink incase you e.g. update or move Xcode for some reason.

File: build/pyutil/file_util.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2012 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import errno
+import os
+
```
Comment (zanderso) [inline build/pyutil/file_util.py:7]: nit: please add comments

**Review by zanderso — APPROVED** [review]

File: build/pyutil/file_util.py (hunk 2 above)
Comment (dnfield) [reply build/pyutil/file_util.py:7]: Done

**Pull request merged**